src/Uns.ts

```diff
@@ -350,90 +395,44 @@ export default class Uns extends NamingService {
         });
       }
       throw new ResolutionError(ResolutionErrorCode.UnspecifiedResolver, {
+        location: data.location,
         domain,
       });
     }
     return data;
   }
 
-  private async getStandardRecords(tokenId: string): Promise<CryptoRecords> {
-    const keys = Object.keys(ResolverKeys.keys);
-    return await this.getMany(tokenId, keys);
-  }
-
-  private async getAllRecords(
-    resolverContract: EthereumContract,
-    tokenId: string,
-  ): Promise<CryptoRecords> {
-    const startingBlock = await this.getStartingBlock(
-      resolverContract,
-      tokenId,
-    );
-    const logs = await this.getNewKeyEvents(
-      resolverContract,
-      tokenId,
-      startingBlock || 'earliest',
-    );
-    const keyTopics = logs.map((event) => event.topics[2]);
-    // If there are no NewKey events we want to check the standardRecords
-    if (keyTopics.length === 0) {
-      return await this.getStandardRecords(tokenId);
-    }
-    return await this.getManyByHash(tokenId, keyTopics);
-  }
-
-  private async getMany(
-    tokenId: string,
-    keys: string[],
-  ): Promise<CryptoRecords> {
-    return (await this.get(tokenId, keys)).records;
-  }
-
-  private async getManyByHash(
-    tokenId: string,
-    hashes: string[],
-  ): Promise<CryptoRecords> {
-    const [keys, values] = (await this.readerContract.call('getManyByHash', [
-      hashes,
-      tokenId,
-    ])) as [string[], string[]];
-    return constructRecords(keys, values);
-  }
-
   private async get(tokenId: string, keys: string[] = []): Promise<DomainData> {
-    const [resolver, owner, values] = await this.readerContract.call(
-      'getData',
-      [keys, tokenId],
-    );
-    return {owner, resolver, records: constructRecords(keys, values)};
-  }
-
-  private isLegacyResolver(resolverAddress: string): boolean {
-    return this.isWellKnownLegacyResolver(resolverAddress);
-  }
-
-  private isWellKnownLegacyResolver(resolverAddress: string): boolean {
-    const legacyAddresses =
-      UnsConfig?.networks[this.network]?.contracts?.Resolver?.legacyAddresses;
-    if (!legacyAddresses || legacyAddresses.length === 0) {
-      return false;
+    const [resultOrErrorL1, resultOrErrorL2] = await Promise.all([
+      this.unsl1.readerContract
```

Comment:
Avoid using contracts directly in Uns class and implement getData method in the UnsInternal class.